libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -61,46 +56,18 @@ class NavigationView : ConstraintLayout, LifecycleObserver {
     private lateinit var tripProgressView: View
     private lateinit var recenterButtonView: View
     private lateinit var routeOverviewButtonView: View
-    private lateinit var mapboxNavigation: MapboxNavigation
     private lateinit var navigationStateViewModel: NavigationStateViewModel
     private lateinit var containerVisibilityViewModel: ContainerVisibilityViewModel
 
-    constructor(context: Context, attrs: AttributeSet?) : this(
-        context,
-        attrs,
-        null,
-        MapInitOptions(context),
-        NavigationViewOptions.Builder().build()
-    )
-
     @JvmOverloads
     constructor(
         context: Context,
         attrs: AttributeSet? = null,
-        accessToken: String?,
         mapInitOptions: MapInitOptions = MapInitOptions(context),
         navigationViewOptions: NavigationViewOptions = NavigationViewOptions.Builder().build()
     ) : super(context, attrs) {
-        this.navigationViewModel = if (accessToken != null) {
-            ViewModelProvider(
-                activity,
-                NavigationViewModelFactory(accessToken, activity.application)
-            )[NavigationViewModel::class.java]
-        } else {
-            val a = context.obtainStyledAttributes(attrs, R.styleable.NavigationView, 0, 0)
-            val attrsAccessToken = a.getString(R.styleable.NavigationView_accessToken)
-            a.recycle()
-            ifNonNull(attrsAccessToken) { token ->
-                ViewModelProvider(
-                    activity,
-                    NavigationViewModelFactory(token, activity.application)
-                )[NavigationViewModel::class.java]
-            } ?: throw IllegalStateException(
-                "Access token must be provided through xml attributes or constructor injection."
-            )
-        }
+
         this.navigationViewOptions = navigationViewOptions
-        this.mapboxNavigation = navigationViewModel.mapboxNavigation
         initializeViewModels()
         this.lifeCycleOwner = context as? LifecycleOwner ?: throw LifeCycleOwnerNotFoundException()
```

Comment:
Should this be replaced with lifecycleOwner from MapboxNavigationApp?

Reply:
NavigationView will have its own `lifecycleOwner` which represents the lifecycle of the `NavigationView`

So no this `lifecycleOwner` is unreleased to `MapboxNavigationApp`